host_modules/systemd_service.py

```diff
@@ -0,0 +1,48 @@
+"""Systemd service handler"""
+
+from host_modules import host_service
+import subprocess
+
+MOD_NAME = 'systemd'
+ALLOWED_SERVICES = ['snmp', 'swss', 'dhcp_relay', 'radv', 'restapi', 'lldp', 'sshd']
+EXIT_FAILURE = 1
+
+
+class SystemdService(host_service.HostModule):
+    """
+    DBus endpoint that executes the service command
+    """
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='s', out_signature='is')
+    def restart_service(self, service):
+        if not service:
+            return EXIT_FAILURE, "Dbus restart_service called with no service specified"
+        if service not in ALLOWED_SERVICES:
+            return EXIT_FAILURE, "Dbus does not support {} service restart".format(service)
```

Comment:
>EXIT_FAILURE [](http://example.com/codeflow?start=19&length=12)

Could gnmi interpret this failure reason and further report NotImplemented error to gnmi_client?

Reply:
GNMI could get this return code and error message.

Reply:
Yes, and I've added this error message interpretation to E2E tests: https://github.com/sonic-net/sonic-mgmt/pull/12478/files